radixdlt-java/src/main/java/org/radix/serialization2/mapper/RadixCBORFactory.java

```diff
@@ -0,0 +1,451 @@
+package org.radix.serialization2.mapper;
+
+//CHECKSTYLE:OFF
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.io.Writer;
+import java.net.URL;
+
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.core.format.InputAccessor;
+import com.fasterxml.jackson.core.format.MatchStrength;
+import com.fasterxml.jackson.core.io.IOContext;
+import com.fasterxml.jackson.dataformat.cbor.CBORConstants;
+import com.fasterxml.jackson.dataformat.cbor.CBORParser;
+import com.fasterxml.jackson.dataformat.cbor.CBORParserBootstrapper;
+
+/**
+ * Factory used for constructing {@link CBORParser} and {@link CBORGenerator}
+ * instances; both of which handle
+ * <a href="https://www.rfc-editor.org/info/rfc7049">CBOR</a>
+ * encoded data.
+ *<p>
+ * Extends {@link JsonFactory} mostly so that users can actually use it in place
+ * of regular non-CBOR factory instances.
+ *<p>
+ * Note on using non-byte-based sources/targets (char based, like
+ * {@link java.io.Reader} and {@link java.io.Writer}): these can not be
+ * used for CBOR documents; attempt will throw exception.
+ * <p>
+ * NOTE: This code has been adapted from the Jackson CBOR class
+ * {@link com.fasterxml.jackson.dataformat.cbor.CBORFactory}.
+ * The only changes have been to change references to
+ * {@code CBORGenerator} to {@link RadixCBORGenerator}.
+ *
+ * @author Tatu Saloranta
```

Comment:
Maybe mention the Radix contributor (you?) here as well.

Also I don't see any licensing information being mentioned here. If not completely mistaken, I think "default copyright" applies then - in favour of Tatu Saloranta or the organisation he represented at the time this code was written. This isn't necessarily compatible with our open source license of the java library.

Could we perhaps have a look at this and either state an Open Source license if such exists for the original file, or find another way to avoid unpleasant legal surprises.

Reply:
Sure, good point. I'll include the Jackson licence link in the file.